File: Baekjoon/단계별/21단계 - 큐, 덱/1966번 - 프린터 큐.cpp
#include <iostream>
#include <queue>
using namespace std;

int t, n, m, cnt;
int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	cin >> t;

	while (t--) {
		// 현재 queue에 m번째에 저장되어 있는 문서가 몇번째로 출력되는지
		cin >> n >> m;
		cnt = 0;
		queue<pair<int, int>> q;
		priority_queue<int> pq;

		for (int i = 0; i < n; i++) {
			int input;
			cin >> input;
			q.push({ i,input });
			pq.push(input);
		}

		while (!q.empty()) {
			int index = q.front().first;
			int value = q.front().second;
			q.pop();

			// 중요도 값 비교
			if (pq.top() == value) {
				pq.pop();
				++cnt;
				if (m == index) {
					cout << cnt << "\n";
					break;
				}
			}
			else
				q.push({ index, value });
		}
	}

	return 0;
}	

File: 백준/Python/Silver/1966번 - 프린터 큐.py
t = int(input())

for _ in range(t):
    n, m = map(int, input().split())
    priority = list(map(int, input().split()))
    index = [i for i in range(n)]
    cnt = 0

    while True:
        # 현재 문서가 중요도가 가장 높다면
        if priority[0] == max(priority):
            cnt += 1
            
            # 궁금한 문서인지 확인
            if index[0] == m:
                print(cnt)
                break
            # 아니라면 리스트에서 제거
            else:
                del priority[0]
                del index[0]
        
        # 나머지 문서들 중 현재 문서보다 중요도가 높은 문서가 하나라도 있으면
        else:
            # Queue의 가장 뒤에 재배치
            priority.append(priority[0])
            del priority[0]
            index.append(index[0])
            del index[0]